dbusutil/dbustest/stub.go

```diff
@@ -0,0 +1,46 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package dbustest
+
+import (
+	"github.com/godbus/dbus"
+)
+
+// stubReadWriteCloser implements ReadWriteCloser for dbus.NewConn
+type stubReadWriteCloser struct{}
+
+func (*stubReadWriteCloser) Read(p []byte) (n int, err error) {
+	return 0, nil
+}
+
+func (*stubReadWriteCloser) Write(p []byte) (n int, err error) {
+	return 0, nil
```

Comment:
I wonder if it should return `len(p), nil`?

Reply:
That does seem reasonable but I haven't tried it. I can do a follow up if it makes sense and doesn't break go-dbus. In general for actual interactions the existing (larger) test connection implements real IO. This is just a minimal do-nothing implementation that can be used when a do-nothing connection is required.

Reply:
Yep, it's most likely fine in existing use cases, the only case something like this could fail in broader context is if the caller checks for the number of bytes written.